hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/impl/HddsDispatcher.java

```diff
@@ -556,11 +557,18 @@ private boolean isContainerFull(Container container) {
     boolean isOpen = Optional.ofNullable(container)
         .map(cont -> cont.getContainerState() == ContainerDataProto.State.OPEN)
         .orElse(Boolean.FALSE);
+    long volumeFreeSpaceToSpare = (long) conf.getStorageSize(
```

Comment:
Please put this new logic into an individual function, say isVolumeFull

Reply:
Done. added a method in VolumeUsage to get this, It now provides both options
1. If percentage is provided via hdds.datanode.volume.min.free.space.percent , the space will be a percent of the (volume capacity-reserved)
2. Else it will fall back to hdds.datanode.volume.min.free.space which if not defined will be 5GB.